src/shared/sam/localLambdaRunner.ts

```diff
@@ -320,7 +321,12 @@ export async function invokeLambdaFunction(
 
         // sam local invoke ...
         const command = new SamCliLocalInvokeInvocation(localInvokeArgs)
+        let samVersion: string | undefined 
         let invokeResult: telemetry.Result = 'Failed'
+
+        // Retrives SAM's version if the CLI exists, otherwise log an error
+        await getSamCliVersion(getSamCliContext()).then(v => { samVersion = v }).catch(err => getLogger().error(err))
```

Comment:
Why is then().catch() needed here, but not in the `createNewSamApp.ts` nor `deploySamApplication.ts` cases?

Reply:
Good point. I moved the `getSamCliVersion` call inside the `try` block since it makes more sense (I did this for all 3 scripts where `samVersion` was added). I also added a missing `catch` block for `resumeCreateNewSampApp` since there wasn't one.